### PR TITLE
Feat/605 build functionality of the progress form bar in the seedlot detail tab

### LINKED ISSUE #605: Build functionality of the progress form bar in the seedlot detail tab
**Describe the task**
Build functionality of the progress form bar in the seedlot detail page, the bar should reflect the current status of the all the form steps for seedlot registration

**Acceptance Criteria**
- [ ] progress bar component correctly tracks function (incomplete, complete, error)
- [ ] shows the users if there are validation errors

### PR DESCRIPTION
**Pull request by craigyu**
# Description
Closes #605

#### New
- NetworkError component for displaying a general network error 
- QueryStatusType for easier TypeScripting with Tanstack Query
- Form progress on the seedlot detail page now display saved status. Note that this does not affect the progress bar inside the form editing screen. 

#### Changed
- None

#### Removed
- None

### How was this tested?
- [ ] 🧠 Not needed
- [x] 👀 Eyeball
- [ ] 🤖 Added tests

###  What gif/image best describes this PR or how it makes you feel?

---
Thanks for the PR!

Any successful deployments (not always required) will be available below.
[Frontend](https://nr-spar-11-frontend.apps.silver.devops.gov.bc.ca/)
[Backend](https://nr-spar-11-backend.apps.silver.devops.gov.bc.ca/actuator/health)
[Oracle-API](https://nr-spar-11-oracle-api.apps.silver.devops.gov.bc.ca/actuator/health)

Once merged, code will be promoted and handed off to following workflow run.
[Main Merge Workflow](https://github.com/bcgov/nr-spar/actions/workflows/merge-main.yml)

### CODE REVIEW

File: frontend/src/components/EmptySection/index.tsx [hunk 1]
```diff
@@ -10,7 +10,7 @@ import './styles.scss';
 interface EmptySectionProps {
   icon?: string;
   title: string;
-  description: string | React.ReactNode;
+  description: React.ReactNode;
```
Comment (mgaseta) [inline frontend/src/components/EmptySection/index.tsx:13]: Nice catch!

File: frontend/src/views/Seedlot/SeedlotDetails/FormProgress/index.tsx [hunk 2]
```diff
@@ -1,29 +1,100 @@
...
+    queryFn: () => getAClassSeedlotProgressStatus(seedlotNumber ?? ''),
+    enabled: getSeedlotQueryStatus === 'success'
+      && (seedlotStatusCode === 'PND' || seedlotStatusCode === 'INC'),
+    refetchOnMount: true,
+    retry: 1
+  });
+
+  useEffect(() => {
+    if (formProgressStatusQuery.status === 'success') {
+      const retreivedProgress = formProgressStatusQuery.data
```
Comment (mgaseta) [inline frontend/src/views/Seedlot/SeedlotDetails/FormProgress/index.tsx:60]: Non-blocking: typo on `retreivedProgress`.

**Review by mgaseta — APPROVED** [review]
Nice job! :1st_place_medal: :100: :rocket: 
Found only one typo.

File: frontend/src/views/Seedlot/SeedlotDetails/FormProgress/index.tsx (hunk 2 above)
Comment (craigyu) [reply frontend/src/views/Seedlot/SeedlotDetails/FormProgress/index.tsx]: Gotta say it again, idk how you catch these

Comment (craigyu) [reply frontend/src/views/Seedlot/SeedlotDetails/FormProgress/index.tsx]: fixed https://github.com/bcgov/nr-spar/pull/811/commits/e0472f37dc2d53093e9c7d949a586cd2442511d9

**Pull request merged**